src/context.ts

```diff
@@ -69,9 +74,56 @@ function getContextFromWorkflow(): Context {
     }
   }
 
-  return context;
+  return {
+    commitDate: await getCommitDateFromWorkflow(context.sha, toolkit),
+    ...context
+  } as Context;
 }
 
 async function getContextFromGit(): Promise<Context> {
-  return await Git.context();
+  const ctx = await Git.context();
+
+  return {
+    commitDate: await Git.commitDate(ctx.sha),
+    ...ctx
+  } as Context;
+}
+
+async function getCommitDateFromWorkflow(sha: string, toolkit: Toolkit): Promise<Date> {
+  const event = GitHub.context.payload as unknown as {
+    // branch push
+    commits?: Array<{
+      timestamp: string;
+      // commit sha
+      id: string;
+    }>;
+    // tags
+    head_commit?: {
+      timestamp: string;
+      // commit sha
+      id: string;
+    };
+  };
+
+  if (event.commits) {
+    const commitDate = event.commits.find(x => x.id === sha)?.timestamp;
+    if (commitDate) {
+      return new Date(commitDate);
+    }
+  }
+
+  if (event.head_commit) {
+    if (event.head_commit.id === sha) {
+      return new Date(event.head_commit.timestamp);
+    }
+  }
+
+  // fallback to github api for commit date
+  const commit = await toolkit.github.octokit.request('GET /repos/{owner}/{repo}/commits/{commit_sha}', {
+    commit_sha: sha,
+    owner: GitHub.context.repo.owner,
+    repo: GitHub.context.repo.repo
+  });
```

Comment:
I don't think this would work for pull requests from a fork

Reply:
see this: https://github.com/trim21/metadata-action/pull/2

Reply:
Looking at https://docs.github.com/en/rest/commits/commits?apiVersion=2022-11-28#get-a-commit--parameters I think we can just pass the full ref like:

```suggestion
  const commit = await toolkit.github.octokit.request('GET /repos/{owner}/{repo}/commits/{GitHub.context.ref}', {
    owner: GitHub.context.repo.owner,
    repo: GitHub.context.repo.repo
  });
```

In case of a pull request it would be smth like `'GET /repos/{owner}/{repo}/commits/refs/pull/471/merge'` for your pr.

Reply:
no, a commit in fork will also always exists in any fork

for example: https://github.com/trim21/metadata-action/commit/4a4aa3e4cba597bb90d7342e9a2da7891cb822eb

Reply:
> see this: [trim21#2](https://github.com/trim21/metadata-action/pull/2)

I meant for a pull request from a fork but seems GitHub handles that: https://github.com/docker/metadata-action/actions/runs/11832905758/job/32970605385?pr=471#step:3:18

Reply:
in anyway, if sha is correct, we will get a commit. even if we send request to head repo not base repo